crates/solver/src/order_balance_filter.rs

```diff
@@ -124,6 +129,31 @@ pub fn balance_orders(
     result
 }
 
+/// Prioritise which orders to account first for using users remaining balance.
+/// Given the external price vector, orders are sorted descending by the
+/// expected surplus (likelyhood of being matchable)
+fn sort_orders_for_balance_priority(orders: &mut Vec<Order>, external_prices: &ExternalPrices) {
+    orders.sort_by(|lhs, rhs| {
```

Comment:
This should probably use `sort_by_cached_key()` for 2 reasons:
1. no duplicated code for computing surplus of `lhs` and `rhs`
2. surplus only gets computed once instead of on every comparison